docs/language/learn-ql/javascript/type-tracking.rst

```diff
@@ -0,0 +1,525 @@
+Tutorial: API modelling using type tracking
+===========================================
+
+This tutorial demonstrates how to build a simple model of the Firebase API in QL
```

Comment:
Hm, I try to avoid inserting links unless I think it's a good time for the reader to click it and I don't want the reader to click it here. It's linked to at the point where the destination of that link becomes relevant, and where it's also mentioned that knowledge of Firebase is not required. Isn't that the better place to have the link?